aten/src/ATen/native/cpu/HistogramKernel.cpp

```diff
@@ -0,0 +1,209 @@
+#include <ATen/native/Histogram.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/native/BucketizationUtils.h>
+
+#include <algorithm>
+#include <mutex>
+#include <tuple>
+
+namespace at { namespace native {
+
+namespace {
+
+const int64_t HISTOGRAM_GRAIN_SIZE = 200;
+
+template<typename input_t, typename output_t, typename ExtractWeight_t, bool LinearBinEdges>
+void histogram_cpu_contiguous(Tensor& hist, const Tensor& bin_edges, const Tensor& input, const Scalar& min, const Scalar& max, ExtractWeight_t extract_weight) {
+    int64_t numel_in = input.numel();
+    if (!numel_in) {
+        return;
+    }
+
+    int64_t numel_be = bin_edges.numel();
+    TORCH_CHECK(numel_be > 0, "expected bin_edges to contain at least 1 element but it contains 0");
+
+    const input_t *data_in = input.data_ptr<input_t>();
+    const input_t *data_be = bin_edges.data_ptr<input_t>();
+
+    output_t *data_out = hist.data_ptr<output_t>();
+
+    input_t min_bound = min.to<input_t>();
+    input_t max_bound = max.to<input_t>();
+
+    if (min_bound == max_bound) {
+        min_bound = input.min().item<input_t>();
+        max_bound = input.max().item<input_t>();
+    }
+    if (min_bound == max_bound) {
```

Comment:
comment here

Reply:
I actually don't see how this can be right. Is this for the "All but the last (righthand-most) bin is half-open" behavior? Why would you subtract one from `min_bound` then?